src/writer/stringBuilder.ts

```diff
@@ -12,8 +12,55 @@ export const convertKeyValueToLine = ({
   let line = "";
 
   if (key) {
-    // If the key contains any special characters, we need to wrap it in quotes
-    line += /^\w*$/.test(key) ? key : JSON.stringify(key);
+    // Check if the key is a valid TypeScript identifier:
+    // 1. Must start with a letter, underscore, or dollar sign
+    // 2. Can contain letters, numbers, underscores, or dollar signs
+    // 3. Cannot be a reserved keyword
+    const isValidTsIdentifier =
+      /^[a-zA-Z_$][a-zA-Z0-9_$]*$/.test(key) &&
+      ![
+        "class",
+        "interface",
+        "type",
+        "enum",
+        "break",
+        "case",
+        "catch",
+        "continue",
+        "debugger",
+        "default",
+        "delete",
+        "do",
+        "else",
+        "finally",
+        "for",
+        "function",
+        "if",
+        "in",
+        "instanceof",
+        "new",
+        "return",
+        "switch",
+        "this",
+        "throw",
+        "try",
+        "typeof",
+        "var",
+        "void",
+        "while",
+        "with",
+        "implements",
+        "package",
+        "protected",
+        "static",
+        "let",
+        "const",
+        "null",
+        "true",
+        "false"
+      ].includes(key);
```

Comment:
lets create a constant for the reserved keywords since they're used twice now, wanna make a `constants.ts` file with this one and any others you are still using?